arangod/IResearch/IResearchViewCoordinator.cpp

```diff
@@ -67,7 +69,47 @@ arangodb::Result IResearchViewCoordinator::appendVelocyPackDetailed(
     );
   }
 
+  // open up a read transaction and add all linked collections to verify that
+  // the current user has access
+
   arangodb::velocypack::Builder links;
+  std::vector<std::string> collections;
+  for (auto& entry: _collections) {
+    collections.emplace_back(entry.second.first);
+  }
+
+  Result result = arangodb::basics::catchToResult([this, &collections]() -> arangodb::Result {
```

Comment:
this check should be for the case when !forPersistance
what is the benefit of using catchToResult over a regular try-catch?

Reply:
More succinct way to do standard exception handling. (Write all the different catch and convert-to-Result cases in one place, use it everywhere.) Main reason not to use it would be if you need to do something non-standard with the exception before it's converted to a Result.